### DISCUSSION TITLE
Post attached files in announcement

### DISCUSSION THREAD

**Discussion by SebasMeza** [ucfopen/canvasapi#620] [Q&A] [answered]
Hi community,

I'm trying to post announcements through canvas API since I do have a lot of courses and the information is the same.

I am doing the following:
```
clase = canvas.get_course(curso)
clase.create_discussion_topic(
    title = anns[msg]["title"],
    message = anns[msg]["message"],
    is_announcement = True,
    attachment = open(files[i], "rb")
    )
```

and also I tried adding a "s" and the end of the attachment parameter:
```
clase = canvas.get_course(curso)
clase.create_discussion_topic(
    title = anns[msg]["title"],
    message = anns[msg]["message"],
    is_announcement = True,
    attachments = open(files[i], "rb")
    )        
```
But I'm getting this error: CanvasException: Encountered an error: status code 500.

I don't know what am I doing wrong or if it is still a capability the package does not have.

I ran the following as a solution and worked and now I am wondering why it didn't work with canvasapi.

```
headers = {"Authorization": "Bearer " + token}
files = {"attachment": open(filename, "rb")}
data = {
    "title": title,
    "message": message,
    "is_announcement":True
}

response = requests.post(API_URL, headers=headers, files=files, data=data)
```

Any help would be appreciated! @bennettscience 

Sebastian

**Comment by Thetwam** [comment]
Hey @SebasMeza, looks like you've found a bug! I've created an issue to address it: #621

**Reply by SebasMeza** [reply]
Excellent! I would be more than happy to help and test!

Hoping to hear from you again soon.


SMeza

**Reply by Thetwam** [reply]
Pull request up with potential fix: #622

**Reply by Thetwam** [reply, marked as answer]
@SebasMeza The fix has been released in CanvasAPI version 3.2.0

Your first code block should work now. Alternatively, you can set `attachment` equal to the filename (as a string) and CanvasAPI will handle opening, reading, and closing the file for you.
